### PR TITLE
My Event Card New UI Design

### PR DESCRIPTION
**Pull request by arnaudlvq**
### Improvements to the "My events" screen

This section concerned the creation of a more attractive user interface for the "event card".

- It now incorporates a visual for the event's {Public, Private} status.
- It's now a stacked card: such a design takes into account small screen resolutions and prevents text from being overwritten.
- I've also added the tag list to quickly remind the user what the event is about.
- The Event now includes a link to a party picture, it has a placeholder in case it is not defined, with a black overlay

### CODE REVIEW

**Comment by arnaudlvq** [comment]
Screenshot_20240509_053031_Chimpagne

**Review by Kuriv2001 — COMMENTED** [review]
Looks nicely done!

I like how you implemented the maxLines and overflow part in the text composable, this should also be done in other parts of our app. Otherwise this could lead to some unfavorable behavior in some cases.

The rest of the PR also seams to cleanly deploy the new approach with the cards. Everything seams to work and looks much better.

Note: Don't forget to create a task for the todo of the dynamic image

**Review by Kuriv2001 — APPROVED** [review]

File: app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEvent.kt [hunk 1]
```diff
@@ -23,7 +23,8 @@ data class ChimpagneEvent(
     val ownerId: ChimpagneAccountUID = "",
     val supplies: Map<ChimpagneSupplyId, ChimpagneSupply> = mapOf(),
     val parkingSpaces: Int = 0,
-    val beds: Int = 0
+    val beds: Int = 0,
+    val image: String = "" // TODO: Add image
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/model/database/ChimpagneEvent.kt:27]: imageURL would be more descriptive

File: app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt [hunk 2]
```diff
@@ -89,13 +99,9 @@ fun MyEventsScreen(navObject: NavigationActions, myEventsViewModel: MyEventsView
...
                   items(uiState.createdEvents.values.toList()) { event ->
-                    ShortEventCard(
-                        title = event.title,
-                        date = timestampToStringWithDateAndTime(event.startsAtTimestamp),
-                        location = event.location.name,
-                        modifier = Modifier.testTag("a created event")) {
-                          navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
-                        }
+                    EventCard(event = event, modifier = Modifier.testTag("a created event")) {
+                      navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt:103]: Choose between string concatenation styles and stick with it, at least on the same line

File: app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt [hunk 3]
```diff
@@ -113,13 +119,9 @@ fun MyEventsScreen(navObject: NavigationActions, myEventsViewModel: MyEventsView
...
                   items(uiState.joinedEvents.values.toList()) { event ->
-                    ShortEventCard(
-                        title = event.title,
-                        date = timestampToStringWithDateAndTime(event.startsAtTimestamp),
-                        location = event.location.name,
-                        modifier = Modifier.testTag("a joined event")) {
-                          navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
-                        }
+                    EventCard(event = event, modifier = Modifier.testTag("a joined event")) {
+                      navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt:123]: same.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt [hunk 4]
```diff
@@ -139,13 +141,9 @@ fun MyEventsScreen(navObject: NavigationActions, myEventsViewModel: MyEventsView
...
                   items(uiState.pastEvents.values.toList()) { event ->
-                    ShortEventCard(
-                        title = event.title,
-                        date = timestampToStringWithDateAndTime(event.startsAtTimestamp),
-                        location = event.location.name,
-                        modifier = Modifier.testTag("past_event_card")) {
-                          navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
-                        }
+                    EventCard(event = event, modifier = Modifier.testTag("past_event_card")) {
+                      navObject.navigateTo(Route.VIEW_DETAIL_EVENT_SCREEN + "/${event.id}")
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/MyEventsScreen.kt:145]: same

**Pull request merged**